mls-rs/src/group/mod.rs

```diff
@@ -851,16 +853,19 @@ where
         &mut self,
         signer: Option<SignatureSecretKey>,
         signing_identity: Option<SigningIdentity>,
+        leaf_node_extensions: Option<ExtensionList>,
     ) -> Result<Proposal, MlsError> {
         // Grab a copy of the current node and update it to have new key material
-        let mut new_leaf_node = self.current_user_leaf_node()?.clone();
+        let mut new_leaf_node: LeafNode = self.current_user_leaf_node()?.clone();
 
+        let new_leaf_node_extensions =
+            leaf_node_extensions.unwrap_or(new_leaf_node.ungreased_extensions());
         let secret_key = new_leaf_node
             .update(
                 &self.cipher_suite_provider,
                 self.group_id(),
                 self.current_member_index(),
-                self.config.leaf_properties(),
+                Some(self.config.leaf_properties(new_leaf_node_extensions)),
```

Comment:
As above - do we ever pass None?

Reply:
At the moment no.

My original plan was to allow passing `None` so that we don't necessarily require new LNEs to be passed but given that the config is a combination of capabilities + LNEs and capabilities are currently driven by the Client's config object I opted to err on the side of preserving the current behavior of ensuring the new capabilities of the client are always set.

I don't see any harm in allowing this internal API to accept an Option but happy to revert it if we prefer.